feat: Add Terra SRPM and appstream page, remove Terra Policy page and move the contents into contributing, and new guidelines page. Also fix some stuff (#196)
FyraLabs/devdocs: pages/terra/guidelines.mdx

---
title: Guidelines
description: "Terra-specific guidelines."
---

import { Callout } from "nextra/components";

# Terra Guidelines

<Callout type="info">
  The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL
  NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED",  "MAY", and
  "OPTIONAL" in this document are to be interpreted as described in
  [RFC 2119](https://www.rfc-editor.org/rfc/rfc2119) and
  [RFC 8174](https://www.rfc-editor.org/rfc/rfc8174).
</Callout>

Packagers and maintainers MUST follow the policies and guidelines listed in this document.

> These are general packaging guidelines for Terra. Everything covered here is one of 3 things:

Reviewer: It's not immediately clear why this is based on differences from the Fedora Packaging Guidelines.

Provide explanation for the structure being based on differences from the Fedora Packaging Guidelines.

Add a link to the Fedora Packaging Guidelines.

Use "Fedora Packaging Guidelines" instead of "Fedora guidelines" to be consistent with Fedora's usage or at least capitalize "guidelines" as a title / proper noun.

- A difference in the Fedora guidelines
- In addition to the Fedora guidelines
- Adding a new guideline

## Terra Packaging Guidelines

### Packager Field

The package maintainer of a package SHOULD be added to the `Packager:{:rpmspec}` preamble as follows:

```rpmspec
Packager: Raboneko <raboneko@fyralabs.com>
```

### Usages of the `mold` linker

We encourage the use of [`mold`](https://github.com/rui314/mold), which may speed up build times especially in large projects.
You MAY enable it in C/++ projects by adding `-fuse-ld=mold` in `CFLAGS`/`CXXFLAGS`.

The `%with_mold` flag is enabled by default in `anda-srpm-macros`. `mold` is also preinstalled in the builders.

You can disable `mold` for Rust and Nim by using `%bcond_with mold`.

### Interpreted Languages (Python, Ruby, etc.)

All packages using interpreted languages follow the traditional Fedora guidelines.
All dependencies **_must_** be packaged individually as they are considered runtime dependencies.

### Dynamically compiled languages (C, C++, Vala)

Dependencies required for runtime MUST be packaged separately.

### Statically compiled languages

> 

Reviewer: Make the capitalization of these titles consistent.

All of these should likely be title-case to be consistent with the rest of the docs and general good practice.

Example:

```rpmspec
%build
%{zig_build_target -r fast -c x86_64_v2} \
  -Demit-docs
```

Note that `-c` also works as a fallback option to the target architecture set by RPM if no following argument is supplied. However, this can **_only_** be used when RPM and Zig's architecture formats are the same, otherwise the build will fail.

Also note that ReleaseFast and especially ReleaseSmall will strip some debug info from the binaries, some Zig projects have enabled build flags to override this and those **_should_** be used if available (this will usually be `-Dstrip=false`). If they are not, you may need to disable debug packages using the global variable `%global debug_package %{nil}` if not enough debug info is preserved to strip.

For projects where dynamic linking of all dependencies is simply not possible (such as the version in the Fedora repos being too new or too old) or that have no dependencies to link, you can pass the `-s` flag. You can (and should) still dynamically link compatible dependencies by using `-fsys=pkgname`.

> ### Shell completions

Reviewer: Make capitalization consistent.

- It is RECOMMENDED to use `%cargo_license_online` and `%cargo_license_summary_online`, although they are not a strict requirement.
- You SHOULD NOT use both `%cargo_build` and `%cargo_install` in the same spec file as `cargo install` might cause a rebuild due to a bug. You SHOULD only include either one of them. In most cases, you can just omit `%cargo_build` entirely and it will just build fine.
Example:

```rpmspec
%prep
%autosetup -n %{crate}-%{version} -p1
%cargo_prep_online

%build
%{cargo_license_online} > LICENSE.dependencies

%install
%cargo_install
```

In some rare cases, you might need to use `%cargo_build` and `%crate_install_bin` instead:

```rpmspec
%prep
%autosetup -n %{crate}-%{version}
%cargo_prep_online

%build
%{cargo_license_online} > LICENSE.dependencies
%{cargo_build} --locked

%install
# install the binary from target/rpm/%{crate} to %buildroot%_bindir
%crate_install_bin
```

If Rust nightly is required, add the `%rustup_nightly` in `%prep`.

Check the [Rust section on the SRPM page](../terra/SRPM#rust) for more information on these macros.

#### Go

> If you use `go2rpm`, you SHOULD NOT use the default naming (except non-leaf packages) but you MAY `Provides:{:rpmspec}`, as an unnecessarily long package name with unnecessary information is bad for UX.

Reviewer 2: ```suggestion
If you use `go2rpm`, you SHOULD NOT use the default naming (except  for non-leaf packages), as an unnecessarily long package name is bad for UX. You MAY put the name generated by `go2rpm` in the Provides:{:rpmspec}` field.
```

# ^ source files, depending on the language, could be in other locations
%_libdir/*.so
%_libdir/*.a
%_libdir/pkgconfig/%{name}.pc
# and other development files (.vapi, .typelib, .gir, etc.).
```

### Miscellaneous

#### Patches

- A package's `%autosetup{:rpmspec}` macro SHOULD use the `-p1{:rpmspec}` flag only if patches are being applied.
- Packages SHOULD apply patches during `%autosetup{:rpmspec}`. If this is not possible (typically when `%autosetup{:rpmspec}` is not used), you SHOULD use [`%autopatch{:rpmspec}`](../rpm/macros#autopatch).

#### Fixing Packages

- If a package fix PR is fixing/changing anything user-facing, the `Release:{:rpmspec}` tag SHOULD be bumped. The only exception SHOULD be if you are fixing a package where the current version does not and has never built.
  - Some (not all!) examples of user-facing fixes requiring a `Release:{:rpmspec}` bump:
    - Installing a missing file.
    - Adding a package to the `Requires:`.
    - Changing any user-facing tags (`URL:{:rpmspec}`, `License:{:rpmspec}`, `Summary:{:rpmspec}`, etc.).
- If a package fix PR is fixing/changing anything build-facing, the `Release:{:rpmspec}` tag SHOULD NOT need to be bumped.

>   - Some (not all!) examples of 'user-facing' fixes requiring a `Release:` bump:

Reviewer 3: ```suggestion
  - Some (not all!) examples of 'user-facing' fixes requiring a `Release:{:rpmspec}` bump:
```

    - Adding a package to the `BuildRequires:{:rpmspec}`.
    - Fixing a package that failed to build when the `Version:{:rpmspec}` tag is bumped.
    - Adding/changing new macros, build steps, install steps, or general source code handing during build time (assuming this does not affect the installed files the user will see).
- If you are unsure if a `Release:{:rpmspec}` bump is needed, please open the PR and ask, and we are happy to provide guidance!

#### Usage of Periods in Package Fields
- You MUST NOT have a period at the end of the `Summary:{:rpmspec}` tag, and you SHOULD have a period at the end of all `%description{:rpmspec}` descriptions.

#### Usage of %elifarch

> To make specs easier to read, you SHOULD use `%elifarch{:rpmspec}` instead of `%endif{:rpmspec}` followed by `%ifarch{:rpmspec}`

Reviewer 3: Words this a bit more clearly.
```suggestion
To make specs easier to read, you SHOULD use `%ifarch{:rpmspec}` followed by `%elifarch{:rpmspec}` before `%endif{:rpmspec}` in specs where build steps are arch dependent.
```

---
title: Guidelines
description: "Terra-specific guidelines."
---

import { Callout } from "nextra/components";

> # Terra Guidelines

Reviewer 4: add something about how contributors SHOULD test built packages functionality,, (and ai policies)

Example:

```rpmspec
%prep
%autosetup -n %{crate}-%{version} -p1
%cargo_prep_online

%build
%{cargo_license_online} > LICENSE.dependencies

%install
%cargo_install
```

In some rare cases, you may need to use `%cargo_build{:rpmspec}` and `%crate_install_bin{:rpmspec}` instead:

```rpmspec
%prep
%autosetup -n %{crate}-%{version}
%cargo_prep_online

%build
%{cargo_license_online} > LICENSE.dependencies
%{cargo_build} --locked

%install
# install the binary from target/rpm/%{crate} to %buildroot%_bindir
%crate_install_bin
```

If Rust nightly is required, add the `%rustup_nightly{:rpmspec}` macro to `%prep{:rpmspec}`.

Check the [Rust section on the SRPM page](../terra/srpm#rust) for more information on these macros.

#### Go
These are guidelines for packaging projects written in [Go](https://go.dev).

For the official Go language documentation, [see here](https://go.dev/doc).

- You SHOULD add (at least!) the following build dependencies to your spec file:
  - `BuildRequires: go-rpm-macros{:rpmspec}`
  - `BuildRequires: go-srpm-macros{:rpmspec}`

> - If you use `go2rpm`, you SHOULD NOT use the default naming (except for non-leaf packages), as an unnecessarily long package name is bad for UX. You MAY put the name generated by `go2rpm` in the `Provides:{:rpmspec}` field.

Reviewer 5: non-leaf?

  - If using `%zig_build_target`, you SHOULD add `BuildRequires: anda-srpm-macros{:rpmspec}` to your spec.
  - For more information on `%zig_build_target`, please see its documentation on our [macros page](../terra/srpm#zig_build_target).

Regardless of build macro in use, they should be used in either the `%build{:rpmspec}` or `%install{:rpmspec}` section depending on how the project is built (as some can be built directly into root by setting the `DESTDIR=%{buildroot}{:rpmspec}` variable).

Example using `%zig_build`:
```rpmspec
%build
%{zig_build} \
  -Demit-docs
```

Example using `%zig_build_target`:
```rpmspec
%build
%{zig_build_target -r fast -c x86_64_v2} \
  -Demit-docs
```

- Note that `ReleaseFast` and especially `ReleaseSmall` will strip some debug info from the binaries.
- Some Zig projects have enabled build flags to override this and those SHOULD be used if available (this will usually be `-Dstrip=false`).
- If they are not, you may need to disable debug packages using the global variable `%global debug_package %{nil}{:rpmspec}` if not enough debug info is preserved to strip.

> For projects where dynamic linking of all dependencies is simply not possible (such as the version in the Fedora repos is too new or too old) or that have no dependencies to link, you MAY use the `-s` flag. You SHOULD still dynamically link compatible dependencies by using `-fsys=pkgname{:rpmspec}`.

Reviewer 3: I didn't clarify this very well. 🤦 
```suggestion
For projects where dynamic linking of all dependencies is simply not possible (such as the version in the Fedora repos is too new or too old) or that have no dependencies to link, you MAY use `%zig_build_target` with the `-s` flag. You SHOULD still dynamically link compatible dependencies by using `-fsys=pkgname{:rpmspec}`.
```

---
title: Guidelines
description: "Terra-specific guidelines."
---

import { Callout } from "nextra/components";

# Terra Guidelines

<Callout type="info">
  The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL
  NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED",  "MAY", and
  "OPTIONAL" in this document are to be interpreted as described in
  [RFC 2119](https://www.rfc-editor.org/rfc/rfc2119) and
  [RFC 8174](https://www.rfc-editor.org/rfc/rfc8174).
</Callout>

> Packagers and maintainers MUST follow the policies and guidelines listed in this document.

Reviewer 6: what's the difference between a packager and maintainer, this isn't defined anywhere

Author: This is not part of this PR

> These are packaging guidelines for Terra. Everything covered here is one of 3 things:

Reviewer 6: ```suggestion
Everything covered here is one of 3 things:
```
feels a bit redundant, but also necessary to have, think about this one

> - A difference in the [Fedora Packaging Guidelines]

Reviewer 6: ```suggestion
- A difference from the [Fedora Packaging Guidelines]
```

- In addition to the [Fedora Packaging Guidelines]

> - Adding a new guideline

Reviewer 6: how is this any different than the line above it??

Author: It's a guideline Fedora doesn't have

Author: Clarified wording

> If something seems missing, please refer to the [Fedora Packaging Guidelines].

Reviewer 6: suggest reading it generally

## Terra Packaging Guidelines

### Packager Field

The package maintainer of a package SHOULD be added to the `Packager:{:rpmspec}` preamble as follows:

```rpmspec
Packager: Raboneko <raboneko@fyralabs.com>
```

> ### Usages of the `mold` Linker

Reviewer 6: ```suggestion
### Usage of the `mold` Linker
```
if you need it to be plural use "Uses" instead, "Usages" is not used very commonly and sounds weird

Author: This is also not part of this PR but it looks like the s may be a typo, removed that

> We encourage the use of [`mold`](https://github.com/rui314/mold), which may speed up build times especially in large projects.

Reviewer 6: ```suggestion
We encourage the use of [`mold`](https://github.com/rui314/mold), which can speed up build times, especially in large projects.
```

You MAY enable it in C/++ projects by adding `-fuse-ld=mold` in `CFLAGS`/`CXXFLAGS`.

The `%with_mold` flag is enabled by default in `anda-srpm-macros`. `mold` is also preinstalled in the builders.

You can disable `mold` for Rust and Nim by using `%bcond_with mold`.

### Interpreted Languages (Python, Ruby, etc.)

> All packages using interpreted languages follow the traditional Fedora guidelines.

Reviewer 6: ```suggestion
All packages using interpreted languages follow the Fedora guidelines.
```

Reviewer 7: should?

Reviewer 6: does that fit with the RFC RFC?

Author: Added SHOULD

> All dependencies **_must_** be packaged individually as they are considered runtime dependencies.

Reviewer 6: this doesn't fit the RFC RFC it looks, as i mentioned before i can't be assed to read it so again an exercise for the author

Author: Read the RFC, or better yet just look at the info callout
<img width="719" height="230" alt="image" src="https://github.com/user-attachments/assets/0a68c2ca-7956-44cb-b887-c40ef05c9b36" />

Reviewer 6: why is it italicized and bolded rather than all caps like every other usage

Author: No idea

### Dynamically Compiled Languages (C, C++, Vala)

> Dependencies required for runtime MUST be packaged separately.

Reviewer 6: why? you explain for python but not here, explain

Author: This is not part of this PR

### Statically Compiled Languages

> As Terra does not strictly follow Fedora's reproducibility requirements, and we do not want large amounts of development libraries for them, packagers SHOULD vendor dependencies on build time.

Reviewer 6: ```suggestion
Terra does not strictly follow Fedora's reproducibility requirements, and we do not want large amounts of development libraries for them, packagers SHOULD vendor dependencies on build time.
```
dependencies that are already packaged can be used instead of vendored ones in some cases, you may wanna note that this can be a preferable solution for some software and avoids duplication.

this also isn't an issue of us not wanting packages, but of packagers being required to package them, maybe focus on how this streamlines adding a package

Author: Not part of this PR, don't want to change more things.

> Terra's mock sandbox has networking enabled, so builders can download the dependencies directly on build time.

Reviewer 6: ```suggestion
Terra's Mock sandbox has networking enabled, builders can download the dependencies directly on build time.
```

- You SHOULD add (at least!) the following build dependencies to your spec file:
  - `BuildRequires: anda-srpm-macros{:rpmspec}`
  - `BuildRequires: cargo-rpm-macros{:rpmspec}`
  - `BuildRequires: rust-srpm-macros{:rpmspec}`
- You MAY generate the spec file using `rust2rpm`.
- You SHOULD NOT use the tradtional Fedora `%cargo_prep{:rpmspec}` macro. Instead, use `%cargo_prep_online{:rpmspec}` from `anda-srpm-macros`. Also, you SHOULD remove the `%generate_buildrequires{:rpmspec}` macro, as it is useless.
- It is RECOMMENDED to use `%cargo_license_online{:rpmspec}` and `%cargo_license_summary_online{:rpmspec}`, although they are not a strict requirement.
- You SHOULD NOT use both `%cargo_build{:rpmspec}` and `%cargo_install{:rpmspec}` in the same spec file as anything that calls `cargo install` might cause a rebuild due to a cargo bug. You SHOULD only include `%cargo_build{:rpmspec}` or `%cargo_install{:rpmspec}`. In most cases, you can just omit `%cargo_build{:rpmspec}` entirely and it will build just fine.

Example:

```rpmspec
%prep
%autosetup -n %{crate}-%{version} -p1
%cargo_prep_online

%build
%{cargo_license_online} > LICENSE.dependencies

%install
%cargo_install
```

> In some rare cases, you may need to use `%cargo_build{:rpmspec}` and `%crate_install_bin{:rpmspec}` instead:

Reviewer 6: ```suggestion
Rarely, you may need to use `%cargo_build{:rpmspec}` and `%crate_install_bin{:rpmspec}` instead:
```

```rpmspec
%prep
%autosetup -n %{crate}-%{version}
%cargo_prep_online

%build
%{cargo_license_online} > LICENSE.dependencies
%{cargo_build} --locked

%install
# install the binary from target/rpm/%{crate} to %buildroot%_bindir
%crate_install_bin
```

If Rust nightly is required, add the `%rustup_nightly{:rpmspec}` macro to `%prep{:rpmspec}`.

Check the [Rust section on the SRPM page](../terra/srpm#rust) for more information on these macros.

#### Go

> These are guidelines for packaging projects written in [Go](https://go.dev).

Reviewer 6: this is kinda obvious and not needed, same for rust's intro above

Reviewer 3: I think this is really overestimating how familiar the average person is with coding languages.

Author: We are keeping the links.

For the official Go language documentation, [see here](https://go.dev/doc).

- You SHOULD add (at least!) the following build dependencies to your spec file:
  - `BuildRequires: go-rpm-macros{:rpmspec}`
  - `BuildRequires: go-srpm-macros{:rpmspec}`
- If you use `go2rpm`, you SHOULD NOT use the default naming, as an unnecessarily long package name is bad for UX. You MAY put the name generated by `go2rpm` in the `Provides:{:rpmspec}` field.
- You SHOULD remove the `%generate_buildrequires{:rpmspec}` section, and add either add `%global gomodulesmode GO111MODULE=on{:rpmspec}` OR use `%goprep_online{:rpmspec}` instead of `%goprep{:rpmspec}` in the `%prep{:rpmspec}` section. 
  - You SHOULD add `BuildRequires: anda-srpm-macros{:rpmspec}` to your spec if using `%goprep_online{:rpmspec}`.

#### Nim

> These are guidelines for packaging projects written in [Nim](https://nim-lang.org).

Reviewer 6: same comment as go and rust

For the official Nim language documentation, [see here](https://nim-lang.org/documentation.html).

- You SHOULD add (at least!) the following build dependencies to your spec file:
  - `BuildRequires: anda-srpm-macros{:rpmspec}`
  - `BuildRequires: nim{:rpmspec}`
- You SHOULD use our [Nim macros](../terra/srpm#nim).

In most cases, use `nimble` to download dependencies in `%prep{:rpmspec}`,
then `%build{:rpmspec}` with `nim`, e.g.[^1]:

```rpmspec
%prep
%autosetup
%nim_prep

%build
%nim_c src/dive

%install
install -Dm755 src/dive -t %buildroot%_bindir
```

The use of the [`atlas`](https://github.com/nim-lang/atlas) package cloner is also recognized, e.g.[^2]:

```rpmspec
%prep
%autosetup -Sgit

%build
atlas init
atlas rep atlas.lock
%nim_c src/netto
```

#### Zig

> These are guidelines for packaging projects written in [Zig](https://ziglang.org).

Reviewer 6: see above, above, and above

    - Installing a missing file.
    - Adding any build or non-build, hard or soft, forward or backward dependencies.
    - Changing any user-facing tags (`URL:{:rpmspec}`, `License:{:rpmspec}`, `Summary:{:rpmspec}`, etc.).
- If a package fix PR is fixing/changing anything build-facing, the `Release:{:rpmspec}` tag SHOULD NOT need to be bumped.
  - Some (not all!) examples changes NOT requiring a `Release:{:rpmspec}` bump:
    - Fixing a package that failed to build when the `Version:{:rpmspec}` tag is bumped.
    - Any changes that do not affect the final package.
- If you are unsure if a bump is needed, bump it.
- You SHOULD add a `%changelog{:rpmspec}` entry for large changes, such as file renames, new build process, etc. Small changes like adding build dependencies or adding one-liners to make a build work MAY be omitted from adding a `%changelog{:rpmspec}` entry.

#### Patches

- Packages SHOULD apply patches (usually via the `-p*` flag) in `%autosetup{:rpmspec}`. If this is not possible (typically when `%autosetup{:rpmspec}` is not used), you SHOULD use [%autopatch](../rpm/macros#autopatch).

> #### Providing your own .desktop file

Reviewer 6: ```suggestion
#### Providing Your Own .desktop File
```

> If you are packaging a graphical application that does not provide its own `.desktop` file, you SHOULD create one.

Reviewer 6: ```suggestion
If you are packaging a graphical app that doesn't provide its own `.desktop` file, you SHOULD create one.
```

> ##### Creating the .desktop file

Reviewer 6: ```suggestion
##### Creating the .desktop File
```

Copy the following template:

```
[Desktop Entry]
Name=PACKAGE NAME
Exec=PROVIDED_BINARY
Icon=ICON NAME
Type=Application
```

> 

Reviewer 6: you may wanna note the WM class field if needed, see my arduino app lab desktop file

Author: What?

The `Icon` tag SHOULD NOT a contain full path, just the name of the icon file.
The `Exec` tag SHOULD be a full path if the binary placed in `/usr/bin` is a symlink.
If this is not the case, it is up to you if you want to use the full path or not (it will look in the user's `$PATH`).

You SHOULD name this file either `package-name.desktop` or `package_app-ID.desktop`.
You MAY add more tags, to see all possible tags, read through the [freedesktop specification](https://specifications.freedesktop.org/desktop-entry/latest/).

> ##### Installing the .desktop file

Reviewer 6: ```suggestion
##### Installing the .desktop File
```

To install the `.desktop` file:

1. Add it to your package's folder
2. Add it as a `Source:{:rpmspec}` in the .spec file
3. Install it
4. List it in the `%files{:rpmspec}` section

It is RECOMMENDED to use the [`%_appsdir` macro](./srpm#_appsdir) from Anda SRPM instead of `%_datadir/applications/`.
With [NeoHtop package] as the example:

```rpmspec
Source1:  NeoHtop.desktop

...

%install
install -Dm644 %{SOURCE1} %{buildroot}%{_appsdir}/NeoHtop.desktop

...

%files
...
%{_appsdir}/NeoHtop.desktop
```

> ##### Validating the .desktop file

Reviewer 6: ```suggestion
##### Validating the .desktop File
```

    - Adding/changing new macros, build steps, install steps, or general source code handing during build time (assuming this does not affect the installed files the user will see).
- If you are unsure if a `Release:{:rpmspec}` bump is needed, please open the PR and ask, and we are happy to provide guidance!

#### Patches

- A package's `%autosetup{:rpmspec}` macro SHOULD use the `-p1{:rpmspec}` flag only if patches are being applied.
- Packages SHOULD apply patches during `%autosetup{:rpmspec}`. If this is not possible (typically when `%autosetup{:rpmspec}` is not used), you SHOULD use [`%autopatch{:rpmspec}`](../rpm/macros#autopatch).

#### Providing your own .desktop file

If you are packaging a graphical application that does not provide its own `.desktop` file, you SHOULD create one.

##### Creating the .desktop file

Copy the following template:

```
[Desktop Entry]
Name=PACKAGE NAME
Exec=PROVIDED BINARY
Icon=ICON NAME
Type=Application
```

The `Icon` and `Exec` tags SHOULD NOT contain full paths, just the name of the icon file/binary to execute.

> You SHOULD name this file either `package-name.desktop` or `package app-ID.desktop`.

Reviewer 4: Imply that you shouldnt use spaces in file names -> `package_app-ID.desktop`. (id put this comment in a diff but the button isnt showing up)

> You MAY add more tags, to see all possible tage, read through the freedesktop specification [here](https://specifications.freedesktop.org/desktop-entry/latest/).

Reviewer 4: - *tags
get rid of redundant word "here"
- [Freedesktop specification](https://specifications.freedesktop.org/desktop-entry/latest/)

##### Installing the .desktop file

> To install the `.desktop` file, add it to your packages folder and add it as a `Source:` in your spec, install it, and list in your `%files`.

Reviewer 4: > To install the `.desktop` file, add it to your packages folder and add it as a `Source:` in your spec, install it, and list in your `%files`.

To install the `.desktop` file:
1. Add it to your package's directory.
2. Add it as a `Source:{:rpmspec}` in the .spec file.
3. Install it
4. List it in the `%files{:rpmspec}` section

not sure how to word it but make 3 more explicit. tell the reader what section to put it in

We will be using the [NeoHtop package] as the example.

> Note that we are using the Anda SRPM [`%_appsdir` macro](./srpm#_appsdir) instead of %_datadir/applications/.

Reviewer 4: should %_datadir/applications/ be in a code snippit?

```rpmspec
Source1:  NeoHtop.desktop

...

%install
install -Dm644 %{SOURCE1} %{buildroot}%{_appsdir}/NeoHtop.desktop

...

%files
...
%{_appsdir}/NeoHtop.desktop
```

##### Validating the .desktop file

You MUST validate your `.desktop` file. To do this, add the following to your spec file:

```rpmspec
BuildRequires:  desktop-file-utils

...

%check
desktop-file-validate %{buildroot}%{_appsdir}/NeoHtop.desktop

...
```

##### Finished Example

> `.desktop` file:

Reviewer 4: name the .desktop properly according to the guidelines for this neohtop thing because its an example. in fact do this whenever the neohtop .desktop file is mentioned. maybe even call back to that step when the reader first reads it?
